app/controllers/api/v3/twilio_sms_delivery_controller.rb

```diff
@@ -7,10 +7,11 @@ def create
 
     twilio_message.update(update_params)
     communication_type = twilio_message.communication.communication_type
-    appointment_id = twilio_message.communication.appointment_id
 
     if communication_type == "missed_visit_whatsapp_reminder" && twilio_message.unsuccessful?
-      AppointmentNotification::Worker.perform_at(Communication.next_messaging_time, appointment_id, "missed_visit_sms_reminder")
+      appointment_reminder_id = twilio_message.communication.appointment_reminder_id
```

Comment:
IDK how long twilio might wait before returning a failed response, so it's possible some of these may have an appointment_reminder_id of nil and the worker will fail. The failed types are failed, undelivered, and unknown. I would guess failed and undelivered happen quickly.  We should probably do a 1off query to determine what the turnaround time is on failed messages, but my guess is that's it's typically very quick.